[inductor] Refactor IR printing

jansel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96024
* #96023

Reland #95567 part 2.  The previous version of this had a bug which that
added test triggers.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

jansel committed:
[inductor] Refactor IR printing

Reland #95567 part 2.  The previous version of this had a bug which that
added test triggers.

[ghstack-poisoned]

jansel committed:
Update on "[inductor] Refactor IR printing"


Reland #95567 part 2.  The previous version of this had a bug which that
added test triggers.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/virtualized.py:
```diff
@@ -96,9 +97,35 @@ def inner(*args):
+        for name, arg in zip(names, args):
+            if arg:
+                lhs = ", ".join(
+                    [
```
This can be a generator instead of a list comprehension

jansel replied:
I don't think it matters since this list will be <10 elements.

jansel committed:
Update on "[inductor] Refactor IR printing"


Reland #95567 part 2.  The previous version of this had a bug which that
added test triggers.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel commented:
@pytorchbot merge

ngimel commented:
cc @desertfire, somehow we didn't catch this memory compression regression in CI, perhaps makes sense to also add SuperSloMo (if it's memory compression is stable enough)

jansel commented:
> @jansel I bisected a memory compression regression on Super_SloMo (0.967 -> 0.8615) to this PR, can you take a look?

Weird.  Perhaps we need to increase `realize_bytes_threshold` because the strings are slightly longer now.
